pkg/beholder/auth.go

```diff
@@ -0,0 +1,23 @@
+package beholder
+
+import "fmt"
+
+// authHeaderKey is the name of the header that the node authenticator will use to send the auth token
+var authHeaderKey = "X-Beholder-Node-Auth-Token"
+
+// authHeaderVersion is the version of the auth header format
+var authHeaderVersion = "1"
+
+// BuildAuthHeaders creates the auth header value to be included on requests.
+// The current format for the header is:
+//
+// <version>:<public_key_hex>:<signature_hex>
+//
+// where the byte value of <public_key_hex> is what's being signed
+func BuildAuthHeaders(signer func([]byte) []byte, pubKey []byte) map[string]string {
+	messageBytes := pubKey
+	signature := signer(messageBytes)
```

Comment:
/nit but do we need to support multiple signing functions? Can we lock the functionality down in this module to save consumers from possibly duplicating/using invalid signing methods?

Reply:
good point, I think realistically we'll just be supporting ed25519 unless something takes the place of CSA keys

Reply:
addressed here 88df7ab02bd80b886d7840ef0161a23b2031e6d8